### DISCUSSION TITLE
Add file explorer style option to select a new location for the blockchain database

### DISCUSSION THREAD

**Discussion by OverActiveBladderSystem** [Chia-Network/chia-blockchain-gui#2456] [Ideas]
"Disk space low, free up space now"

As the blockchain database grows more and more users run into this issue, when its their main OS drive that can impact the OS and all other installed apps once the drive runs out of space.

Inexperienced users struggle to find the hidden (Windows) user directory where the database is located to move it, they also struggle to edit the config yaml and aren't sure what to put for the path, adding a GUI option to define a new folder and/or drive for the blockchain database would help them and lower support requests.

Side quest, when the user chooses a new folder/drive to use, move the existing database to that location with warnings that it should be on an SSD and not a HDD, as some users are unaware that can cause issues when blocks are full and seeks are high.

**Comment by ChiaMineJP** [comment]
Making sense.
The code changes seems a little bit significant, since moving full_node data and other configs requires to be done with all chia services stopped. (The GUI assumes that all of chia services keep running and if any of them stopps it tries to restart the service)
But your idea will help a certain number of users for sure.
I'll note you suggestion.

**Reply by OverActiveBladderSystem** [reply]
Yeah, knowing the database is live with the GUI open its not going to be as straight forward as a dropdown that just instantly works (even if thats the dream), going to be more of a complete closeout and restart like the "wallet resync" which probably has layers of things going on just to do that... but definitely going to continue seeing more inexperienced GUI users with this issue over time who struggle with basics like editing a yaml correctly without wrap/fonts/etc once they find its hidden location.
